File: src/game/tools/rpc/output/wrap/ProductInfoWrap.go
package wraps

import (
	rpc "tipcat.com/common/rpc/simple"
)

type ProductInfoWrap struct {
	
	ProductID string // 商品ID
	SortID int16 // 排序ID
	ProductName string // 商品名
	Desc string // 描述
	Visible bool // 是否可见
	Hot bool // 是否热销
	MoneyType int32 // 货币类型
	MoneyCnt int32 // 发货数量
	GiftMoneyCnt int32 // 赠送数量
	Price float32 // 购买价格
	Currency string // 货币类型
	rpc.Wrapper
}

func (w *ProductInfoWrap)Decode(pck *rpc.Packet) rpc.Wrapper {
	
	w.ProductID = pck.PopString(); 
	w.SortID = pck.PopInt16(); 
	w.ProductName = pck.PopString(); 
	w.Desc = pck.PopString(); 
	w.Visible = pck.PopBool(); 
	w.Hot = pck.PopBool(); 
	w.MoneyType = pck.PopInt32(); 
	w.MoneyCnt = pck.PopInt32(); 
	w.GiftMoneyCnt = pck.PopInt32(); 
	w.Price = pck.PopFloat32(); 
	w.Currency = pck.PopString(); 
	return w
}

func (w *ProductInfoWrap)Encode(pck *rpc.Packet) {
	
	pck.PutString(w.ProductID) 
	pck.PutInt16(w.SortID) 
	pck.PutString(w.ProductName) 
	pck.PutString(w.Desc) 
	pck.PutBool(w.Visible) 
	pck.PutBool(w.Hot) 
	pck.PutInt32(w.MoneyType) 
	pck.PutInt32(w.MoneyCnt) 
	pck.PutInt32(w.GiftMoneyCnt) 
	pck.PutFloat32(w.Price) 
	pck.PutString(w.Currency) 
}

File: src/game/tools/rpc/output/wrap/ProductInfoWrap.cs
using System;
using Common.Net.Simple;

namespace Game 
{
	public class ProductInfoWrap : IWrapper
	{	
		public string ProductID; // 商品ID
		public short SortID; // 排序ID
		public string ProductName; // 商品名
		public string Desc; // 描述
		public bool Visible; // 是否可见
		public bool Hot; // 是否热销
		public int MoneyType; // 货币类型
		public int MoneyCnt; // 发货数量
		public int GiftMoneyCnt; // 赠送数量
		public float Price; // 购买价格
		public string Currency; // 货币类型
		
		public void Decode(Packet pck)
		{	
			ProductID = pck.GetString(); 
			SortID = pck.GetShort(); 
			ProductName = pck.GetString(); 
			Desc = pck.GetString(); 
			Visible = pck.GetBool(); 
			Hot = pck.GetBool(); 
			MoneyType = pck.GetInt(); 
			MoneyCnt = pck.GetInt(); 
			GiftMoneyCnt = pck.GetInt(); 
			Price = pck.GetFloat(); 
			Currency = pck.GetString(); 
		}
		
        public void Encode(Packet pck)
        {	
        	pck.PutString(ProductID); 
        	pck.PutShort(SortID); 
        	pck.PutString(ProductName); 
        	pck.PutString(Desc); 
        	pck.PutBool(Visible); 
        	pck.PutBool(Hot); 
        	pck.PutInt(MoneyType); 
        	pck.PutInt(MoneyCnt); 
        	pck.PutInt(GiftMoneyCnt); 
        	pck.PutFloat(Price); 
        	pck.PutString(Currency); 
        }
	}
}